includes/Integrations/Wcfm/OrderMigrator.php

```diff
@@ -0,0 +1,416 @@
+        }
+
+        $admin_commission = 0;
+
+        foreach ( $commissions as $com ) {
+            $admin_commission += $com['admin_commission'];
+        }
+
+        $dokan_order_data = [
+            'commission_data'         => $commissions,
+            'order_total'             => $order_total,
+            'net_sale'                => $net_amount,
+            'admin_commission_amount' => $admin_commission,
+        ];
+
+        return $dokan_order_data;
+    }
+
+    /**
+     * Process refund for a child order.
+     *
+     * @since DOKAN_MIG_SINCE
+     *
+     * @param WC_Order $child_order
+     * @param integer $seller_id
+     * @param boolean $from_suborder
+     *
+     * @return void
+     */
+    public function process_refund( $child_order, $seller_id, $from_suborder = true ) {
+        // On complete Commission table update
+        $refund_infos = $this->get_refund_requests( $seller_id, $this->order_id );
+
+        if ( empty( $refund_infos ) ) {
+            return;
+        }
+
+        foreach ( $refund_infos as $refund_info ) {
+            $item_id           = absint( $refund_info->item_id );
+            $vendor_id         = absint( $refund_info->vendor_id );
+            $order_id          = absint( $refund_info->order_id );
+            $commission_id     = absint( $refund_info->commission_id );
+            $refunded_amount   = (float) $refund_info->refunded_amount;
+            $refund_id         = (float) $refund_info->ID;
+            $c_refunded_amount = $refunded_amount;
+            $c_refunded_qty    = absint( $this->get_refund_meta( $refund_id, 'refunded_qty' ) );
+            $refund_reason     = $refund_info->refund_reason;
+            $is_partially_refunded = $refund_info->is_partially_refunded;
+            $is_refunded = 0;
+
+            // Create WC Refund Item
+            if ( $order_id ) {
+                $line_item = new \WC_Order_Item_Product( $item_id );
+
+                $c_refunded_qty && $line_item->get_quantity() === $c_refunded_qty ? $is_partially_refunded = 0 : '';
+                ! $is_partially_refunded ? $is_refunded = 1 : '';
+
+                $order = $child_order;
+
+                // API Refund Check
+                $api_refund = $refund_info->refund_status === 'completed' ? true : false;
+
+                $restock_refunded_items = 'true';
+                $refund_tax = $this->get_refund_meta( $refund_id, 'refunded_tax' );
+                $refund_tax = maybe_unserialize( $refund_tax );
+
+                $refund_tax_total = 0;
+
+                if ( $is_refunded ) {
+                    $product = $line_item->get_product();
+
+                    if ( ! empty( $refund_tax ) && is_array( $refund_tax ) ) {
+                        if ( isset( $refund_tax['total'] ) ) {
+                            $refund_tax_total = $refund_tax['total'];
+                        }
+                        if ( ! empty( $refund_tax ) && is_array( $refund_tax ) ) {
```

Comment:
Why this condition is necessary? This block is already inside the same condition. I guess it should be like below:
```
                    if ( ! empty( $refund_tax ) && is_array( $refund_tax ) ) {
                        if ( isset( $refund_tax['total'] ) ) {
                            $refund_tax_total = $refund_tax['total'];
                        } else {
                            foreach ( $refund_tax as $refund_tax_id => $refund_tax_price ) {
                                $refund_tax_total += (float) $refund_tax_price;
                            }
                        }
                    }
```
Kindly verify.